src/control/server/instance.go

```diff
@@ -229,6 +228,10 @@ func (srv *IOServerInstance) NotifyReady(msg *srvpb.NotifyReadyReq) {
 	srv.log.Debugf("I/O server instance %d ready: %v", srv.Index, msg)
 
 	go func() {
+		// TODO KJ: What should we do if the message offers no server socket?
+		// Activate the dRPC client connection to this iosrv
+		srv.drpcClient = drpc.NewClientConnection(msg.DrpcListenerSock)
```

Comment:
I wouldn't put this assignment in the goroutine. The reason this goroutine is here is to avoid any possibility of a deadlock on the channel (reader/writer in same goroutine).

As for what to do if the message doesn't include a socket -- I'm not sure there's much we can do aside from maybe logging it as an error and returning early. I don't think we'd want to signal that the instance is ready because then we'd have a running instance with no way to talk to it. The end result is that you'd get a logged error and then daos_server would hang, waiting for the instances to be ready. We could add a timeout on the part that is waiting to receive the ready notifications, maybe?

Reply:
Thinking about it a bit more... I don't think we should mess around with arbitrary startup timeouts. It could be entirely reasonable for startup to take a while in very large clusters. I think logging an error and returning early here is reasonable, and we just need to sanity-check the message contents in srvModule.handleNotifyReady() before calling this method.

Reply:
Sounds good to me. I was thinking of maybe falling back to previous behavior, using a default socket name, but I guess that doesn't make sense. The daos_server shouldn't start older versions of the IO server.

Reply:
Yeah, I'm inclined to avoid ambiguous behavior as much as possible, and at this point I don't see any reason to not assume that both processes will be running the same version.